### PR TITLE
py36+: drop python3.5 in CI and setup.cfg

### PR DESCRIPTION
**Pull request by asottile**
this is the first part to #7808

I have ~13 other patches that can be previewed in #7811

### CODE REVIEW

File: changelog/7808.improvement.rst [hunk 1]
```diff
@@ -0,0 +1 @@
+pytest now supports python3.6+ only.
```
Comment (nicoddemus) [inline changelog/7808.improvement.rst:1]: Should we consider this a breaking change perhaps?

Comment (asottile) [reply changelog/7808.improvement.rst:1]: I was on the fence, since we're going to release this in a minor version 🤔

Comment (nicoddemus) [reply changelog/7808.improvement.rst:1]: Yeah, but I think there's precedence: we removed a bunch of features in 6.1, and also [5.4](https://docs.pytest.org/en/stable/changelog.html#pytest-5-4-0-2020-03-12).

Comment (asottile) [reply changelog/7808.improvement.rst]: done!

**Review by nicoddemus — APPROVED** [review]
Thanks!

**Pull request merged**